projects/planner/src/app/learn/learn.component.ts

```diff
@@ -189,7 +176,7 @@ export class LearnComponent implements OnInit {
     }
 
     public isCategorySelected(topicName: string, categoryName: string): boolean {
-        return this.selectedTopic === topicName && this.selectedCategoryName === categoryName;
+        return this.selectedTopic.topic === topicName && this.selectedCategory.name === categoryName;
```

Comment:
You no longer need check by name, since you are able to send the topic and category directly from the UI.